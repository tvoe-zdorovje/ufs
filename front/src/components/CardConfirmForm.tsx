import React from 'react'
import {
  Form,
  Icon,
  Input,
  Button,
  Row,
  Col
} from 'antd'

const FormItem = Form.Item

export interface CardConfirmFormProps {
  confirmCard(params: {number: string, pin: string}, onComplete: Function),
  onComplete(),
  card: any,
  form?: any,
}

export interface CardDetails {
  number: string,
  pin: string,
}

@Form.create<any>()
export default class CardConfirmForm extends React.Component<CardConfirmFormProps, any> {
  handleSubmit = (e) => {
    e.preventDefault()
    this.props.form.validateFields((err) => {
      if (!err) {
        /* tslint:disable-next-line */
        const { number, pin } = this.props.form.getFieldsValue() as CardDetails
        this.props.confirmCard({ number, pin }, this.props.onComplete)
      }
    })
  }

  render() {
    const { getFieldDecorator } = this.props.form
    const card = this.props.card
    return (
      <div>
        <Row>
          <Col span={8} />
          <Col span={8}>
            <Form onSubmit={this.handleSubmit} className="login-form">
              <FormItem>
                {getFieldDecorator('number', {
                  rules: [{ required: true, message: 'Введите номер карты' },
                    {
                      pattern: new RegExp('^[0-9]{16}'),
                      message: 'Номер карты должен состоять из 16 символов 0-9',
                    }],
                  validateTrigger: 'onSubmit',
                  initialValue: card.number,
                })(
                  <Input placeholder="Номер карты" />
                )}
              </FormItem>
              <FormItem>
                {getFieldDecorator('pin', {
                  rules: [{ required: true, message: 'Введите ПИН-код' },
                    {
                      pattern: new RegExp('^[0-9]{4}'),
                      message: 'ПИН-код должен состоять из 4 символов 0-9',
                    }],
                  validateTrigger: 'onSubmit',
                })(
                  <Input
                    prefix={<Icon type="lock" style={{ fontSize: 13 }} />}
                    type="password" placeholder="ПИН-код"
                  />
                )}
              </FormItem>
              <FormItem>
                <Button type="primary" htmlType="submit">
                  Отправить
                </Button>
              </FormItem>
            </Form>
          </Col>
          <Col span={8} />
        </Row>
      </div>
    )
  }
}
