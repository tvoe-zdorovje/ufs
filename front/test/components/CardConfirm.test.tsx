import React from 'react'
import {
  render,
  shallow
} from 'enzyme'
import toJson from 'enzyme-to-json'

import CardConfirmForm, { CardConfirmFormProps } from 'components/CardConfirmForm'

describe('Компонент CardConfirm', () => {
  const number = '1111111111'
  const pin = '1111'
  const card = { number, pin }
  const confirmCard = () => {
  }

  const props: CardConfirmFormProps = {
    onComplete: () => {
    },
    confirmCard,
    card,
  }

  const wrapper = shallow(<CardConfirmForm {...props} />)
  const renderWrapper = render(<CardConfirmForm {...props} />)

  it('использует свойства <CardConfirmForm />', () => {
    expect(wrapper.prop('card')).toBe(card)
  })

  it('отображает компонент Form', () => {
    const html = wrapper.html()
    expect(html.match(/class="ant-form\sant-form-horizontal\slogin-form"/g) || [])
      .toHaveLength(1)
  })

  it('отображает 3 компонента FieldItem', () => {
    const html = wrapper.html()
    expect(html.match(/"ant-row\sant-form-item"/g) || [])
      .toHaveLength(3)
  })

  it('совпадает со своим снимком', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot()
  })
})
