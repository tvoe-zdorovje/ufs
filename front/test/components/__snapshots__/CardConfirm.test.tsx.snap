// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент CardConfirm совпадает со своим снимком 1`] = `
<div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col-8"
    />
    <div
      class="ant-col-8"
    >
      <form
        class="ant-form ant-form-horizontal login-form"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control has-success"
            >
              <input
                class="ant-input ant-input-lg"
                data-__meta="[object Object]"
                id="number"
                placeholder="Номер карты"
                type="text"
                value="1111111111"
              />
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control "
            >
              <span
                class="ant-input-affix-wrapper"
              >
                <span
                  class="ant-input-prefix"
                >
                  <i
                    class="anticon anticon-lock"
                    style="font-size:13px;"
                  />
                </span>
                <input
                  class="ant-input ant-input-lg"
                  data-__meta="[object Object]"
                  id="pin"
                  placeholder="ПИН-код"
                  type="password"
                  value=""
                />
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control "
            >
              <button
                class="ant-btn ant-btn-primary ant-btn-lg"
                type="submit"
              >
                <span>
                  Отправить
                </span>
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="ant-col-8"
    />
  </div>
</div>
`;
